### PR TITLE
Remove Autotools packages from CI (and depends doc)

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a follow-up to https://github.com/bitcoin/bitcoin/pull/30752 and addresses https://github.com/bitcoin/bitcoin/pull/30752#discussion_r1758594864.

### CODE REVIEW

File: depends/packages.md [hunk 1]
```diff
@@ -152,16 +152,9 @@ Most autotools projects can be properly staged using:
 
 ## Build outputs:
 
-In general, the output of a depends package should not contain any libtool
-archives. Instead, the package should output `.pc` (`pkg-config`) files where
+In general, the package should output `.pc` (`pkg-config`) files where
```
Comment (fanquake) [inline depends/packages.md:155]: I don't think this is correct. Given we still have depends packages that can output libtool archives (which we currently remove).

Comment (hebasto) [reply depends/packages.md]: Thanks! Reverted back.

**Comment by hebasto** [comment]
About to close it in favour of https://github.com/bitcoin/bitcoin/pull/30875.

**Comment by fanquake** [comment]
> About to close it in favour of https://github.com/bitcoin/bitcoin/pull/30875.

Mark as draft for now, or close?

**Comment by Sjors** [comment]
Maybe rename the PR to: "Remove Autotools packages from CI (and depends doc)"

**Comment by hebasto** [comment]
> Maybe rename the PR to: "Remove Autotools packages from CI (and depends doc)"

Renamed.

**Comment by kevkevinpal** [comment]
> I think it is fine to do the CI/libtool changes separate. Looks like the other pull is still a draft/WIP anyway.

yup agreed I can remove overlapping changes from this PR and https://github.com/bitcoin/bitcoin/pull/30875

**Comment by kevkevinpal** [comment]
ACK [7a8a6a0](https://github.com/bitcoin/bitcoin/pull/30902/commits/7a8a6a06676dcb4066cc81a4e6872281a09bb00d)

**Review by jonatack — COMMENTED** [review]
LGTM at 7a8a6a06676dcb4066cc81a4e6872281a09bb00d, this patch appears to correctly remove the packages that can be dropped while leaving a few remaining ones that are unrelated.

The changes are small enough that I found it easier to review them together and could be squashed.

**Pull request merged**